orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/job/JobRunner.java

```diff
@@ -0,0 +1,37 @@
+package com.netflix.spinnaker.orca.clouddriver.tasks.job;
+
+import com.netflix.spinnaker.orca.api.pipeline.models.StageExecution;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Deployments of server groups vary wildly across cloud providers. A JobRunner is a cloud-provider
+ * specific way to hook into the Orca infrastructure.
+ */
+public interface JobRunner {
+  /**
+   * @return a list of operation descriptors. Each operation should be a single entry map keyed by
+   *     the operation name, with the operation map as the value.
+   */
+  List<Map> getOperations(StageExecution stage);
+
+  /** @return any additional values that should be included in task outputs */
+  Map<String, Object> getAdditionalOutputs(StageExecution stage, List<Map> operations);
+
+  /** @return true if the resulting value from the Kato call should be used. */
+  boolean isKatoResultExpected();
+
+  /** @return The cloud provider type that this object supports. */
+  String getCloudProvider();
+
+  /**
+   * @return the timeout in millis, if any, of the job so we can set the timeout of the task to the
+   *     same
+   */
+  default Optional<Long> getJobTimeoutMs(StageExecution stage) {
```

Comment:
can you stick a `@Nonnull` on this return value

Reply:
also maybe worth making this `Optional<Duration>` and ditching all the suffixes of various time formats. The value if populated in the stage could use the duration syntax (PT1S etc) from Duration.parse

Reply:
yeah, that's fair. i should've just done that upfront... fixing up!